compiler/cpp/src/thrift/generate/t_rs_generator.cc

```diff
@@ -2722,7 +2756,30 @@ string t_rs_generator::rust_namespace(t_service* tservice) {
 }
 
 string t_rs_generator::rust_struct_name(t_struct* tstruct) {
-  return rust_camel_case(tstruct->get_name());
+  string base_struct_name(rust_camel_case(tstruct->get_name()));
+  if (is_reserved(base_struct_name)) {
+    return "Res" + base_struct_name;
+  } else {
+    return base_struct_name;
+  }
+}
+
+string t_rs_generator::rust_field_name(t_field* tfield) {
+  string base_field_name(rust_snake_case(tfield->get_name()));
+  if (is_reserved(base_field_name)) {
+    return "res_" + base_field_name;
```

Comment:
I agree. I've changed this as well - perhaps you were looking at an older version of the PR?